### ISSUE TITLE
dhcpcd 9.1.1_1 using 100% CPU

### ISSUE THREAD

**Issue by AlaskanEmily** [void-linux/void-packages#22812]
### System

* xuname:  
  Void 5.4.46_1 x86_64 GenuineIntel notuptodate rFFF
* package:  
  dhcpcd-9.1.1_1

### Expected behavior
dhcpcd shouldn't be using 100% CPU at idle.

### Actual behavior
Uses 100% cpu idling. Running with -B doesn't seem to show any problems. Downgrading from 9.1.1_1 to 9.0.2_1 fixes this.

### Steps to reproduce the behavior
Enable dhcpcd. I do also have wpa_supplicant enabled as a dhcp hook, and I am using it on a wireless interface.

**Comment by jpmenil** [comment]
strace on the process will help to show what's going on.

**Comment by abenson** [comment]
Can you try building 9.1.2 (#22928) and see if that still has the issue?

**Comment by AlaskanEmily** [comment]
I am having some difficulty trying 9.1.2

I built dhcpcd 9.1.2 by cloning void-packages and then cherry-picking the PR onto it. I did `./xbps-src binary-bootstrap` followed by `./xbps-src pkg dhcpcd`, which shows the right version of 9.1.2, however after installing with `xbps-install --repository hostdir/binpkg dhcpcd` (which also confirmed the version), running dhcpcd just repeats `ps_ctl_dispatch: cannot handle another client` forever. I made sure the service was stopped (and did a pgrep for any additional dhcp processes just to see), and it just repeats this forever and I never get a lease.

**Comment by abenson** [comment]
Is this wired or wireless?

**Comment by AlaskanEmily** [comment]
Wireless.

**Comment by SamuelSarle** [comment]
I had the same problem with dhcpcd 9.1.1_1 taking up high CPU.
Adding to `/etc/dhcpcd.conf` the line `logfile /var/log/dhcpcd.log` prints upon booting:
```
Jun 30 11:49:20 [961]: dhcpcd-9.1.1 starting
Jun 30 11:49:20 [966]: sending commands to dhcpcd process
Jun 30 11:49:20 [985]: dev: loaded udev
Jun 30 11:49:20 [984]: DUID 00:04:5b:67:c0:81:52:dc:11:cb:b7:8b:d1:4b:85:fe:a1:54
Jun 30 11:49:21 [1005]: sending commands to dhcpcd process
Jun 30 11:49:22 [984]: no interfaces have a carrier
Jun 30 11:49:22 [1013]: sending commands to dhcpcd process
Jun 30 11:49:22 [984]: enp0s25: waiting for carrier
Jun 30 11:49:22 [984]: wlp3s0: waiting for carrier
Jun 30 11:49:22 [984]: wlp3s0: carrier acquired
Jun 30 11:49:22 [984]: wlp3s0: IAID 4b:67:c8:a0
Jun 30 11:49:22 [984]: wlp3s0: adding address fe80::9cbd:fe33:6434:8854
Jun 30 11:49:22 [984]: wlp3s0: carrier lost
Jun 30 11:49:22 [984]: wlp3s0: deleting address fe80::9cbd:fe33:6434:8854
Jun 30 11:49:22 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:22 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:23 [1028]: sending commands to dhcpcd process
Jun 30 11:49:24 [1044]: sending commands to dhcpcd process
Jun 30 11:49:24 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:25 [1054]: sending commands to dhcpcd process
Jun 30 11:49:25 [984]: wlp3s0: carrier acquired
Jun 30 11:49:25 [984]: wlp3s0: connected to Access Point `REDACTED'
Jun 30 11:49:25 [984]: wlp3s0: IAID 4b:67:c8:a0
Jun 30 11:49:25 [984]: wlp3s0: adding address fe80::faac:a34d:402a:9292
Jun 30 11:49:25 [984]: wlp3s0: soliciting an IPv6 router
Jun 30 11:49:26 [1072]: sending commands to dhcpcd process
Jun 30 11:49:26 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:27 [984]: wlp3s0: rebinding lease of 192.168.1.101
Jun 30 11:49:27 [984]: wlp3s0: probing address 192.168.1.101/24
Jun 30 11:49:27 [1102]: sending commands to dhcpcd process
Jun 30 11:49:28 [1105]: sending commands to dhcpcd process
Jun 30 11:49:28 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:29 [1107]: sending commands to dhcpcd process
Jun 30 11:49:30 [1109]: sending commands to dhcpcd process
Jun 30 11:49:30 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:31 [1111]: sending commands to dhcpcd process
Jun 30 11:49:32 [984]: wlp3s0: leased 192.168.1.101 for 7200 seconds
Jun 30 11:49:32 [984]: wlp3s0: adding route to 192.168.1.0/24
Jun 30 11:49:32 [984]: wlp3s0: adding default route via 192.168.1.1
Jun 30 11:49:32 [1129]: sending commands to dhcpcd process
Jun 30 11:49:32 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:33 [1147]: sending commands to dhcpcd process
Jun 30 11:49:34 [1148]: sending commands to dhcpcd process
Jun 30 11:49:34 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:35 [1165]: sending commands to dhcpcd process
Jun 30 11:49:36 [1181]: sending commands to dhcpcd process
Jun 30 11:49:36 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:37 [1200]: sending commands to dhcpcd process
Jun 30 11:49:38 [1208]: sending commands to dhcpcd process
Jun 30 11:49:38 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:39 [984]: wlp3s0: no IPv6 Routers available
Jun 30 11:49:39 [1226]: sending commands to dhcpcd process
Jun 30 11:49:40 [1254]: sending commands to dhcpcd process
Jun 30 11:49:40 [984]: control command: dhcpcd -B wlp3s0
Jun 30 11:49:41 [1284]: sending commands to dhcpcd process
Jun 30 11:49:42 [1358]: sending commands to dhcpcd process
Jun 30 11:49:42 [984]: control command: dhcpcd -B wlp3s0
```
With the `sending commands to dhcpcd process` looping indefinitely.

I did the following to fix:
```
git clone git://github.com/void-linux/void-packages.git
cd void-packages/
./xbps-src binary-bootstrap
./xbps-src install dhcpcd
./xbps-src pkg dhcpcd
sudo xbps-install -f --repository hostdir/binpkgs dhcpcd
sudo sv restart dhcpcd
```

I'm not familiar with the tooling in this repo but the patch from #22928 is contained inside `srcpkgs/dhcpcd/patches/dhcpcd-9.1.1-typo_fix.patch`, and before the `./xbps-src pkg dhcpcd` the fix contained in the patch is clearly applied in `masterdir/builddir/dhcpcd-9.1.1/src/if-options.c`.
The version however remains the same:
```
$ dhcpcd --version
dhcpcd 9.1.1
Copyright (c) 2006-2020 Roy Marples
Compiled in features: INET ARP ARPing IPv4LL INET6 DHCPv6 AUTH PRIVSEP
```

Upon `sv restart dhcpcd`, CPU usage dropped immediately. `dhcpcd.log` contains messages for startup and the looping part doesn't appear anymore.

**Comment by sgn** [comment]
`dhcpcd` had been bumped to `9.1.4`, please update and check if problems still persits.

**Comment by AlaskanEmily** [comment]
It is solved for me with `9.1.4`. I'm not sure if you want to leave the issue open and wait for more responses or not.

**Comment by sgn** [comment]
There was a bug in dhcpcd 9.1.1 related to this problem. I think we're fine now.
